infoschema/infoschema.go

```diff
@@ -624,21 +624,38 @@ func (is *SessionTables) schemaTables(schema model.CIStr) *schemaTables {
 type SessionExtendedInfoSchema struct {
 	InfoSchema
 	LocalTemporaryTables *SessionTables
+	MdlTables            *SessionTables
 }
 
 // TableByName implements InfoSchema.TableByName
 func (ts *SessionExtendedInfoSchema) TableByName(schema, table model.CIStr) (table.Table, error) {
-	if tbl, ok := ts.LocalTemporaryTables.TableByName(schema, table); ok {
-		return tbl, nil
+	if ts.LocalTemporaryTables != nil {
```

Comment:
Before there is no need to check whether LocalTemporaryTables is nil, why now it is necessary?

Reply:
It can be nil now.